testsuite/MDAnalysisTests/topology/test_top.py

```diff
@@ -45,35 +51,310 @@ def test_attr_size(self, top):
+                              (138, 137, 135, 155), (139, 137, 135, 155),
+                              (140, 137, 135, 155))
+    atom_zero_improper_values = ()
+    atom_i_improper_values = ((131, 135, 133, 134), (135, 157, 155, 156))
 
 
 class TestPRM2(TOPBase):
     ref_filename = PRMpbc
     expected_n_atoms = 5071
     expected_n_residues = 1686
     ref_proteinatoms = 22
+    expected_n_bonds = 5070
+    expected_n_angles = 36
+    expected_n_dihedrals = 41
+    expected_n_impropers = 4
     guessed_attrs = ['elements']
+    atom_i = 14
+    expected_n_zero_bonds = 1
+    expected_n_i_bonds = 3
+    expected_n_zero_angles = 3
+    expected_n_i_angles = 8
+    expected_n_zero_dihedrals = 2
+    expected_n_i_dihedrals = 18
+    expected_n_zero_impropers = 0
+    expected_n_i_impropers = 2
+    atom_zero_bond_values = ((0, 1),)
+    atom_i_bond_values = ((14, 15), (14, 16), (8, 14))
+    atom_zero_angle_values = ((0, 1, 2), (0, 1, 3), (0, 1, 4))
+    atom_i_angle_values = ((15, 14, 16), (14, 16, 18), (10,  8, 14),
+                           (8, 14, 15), (8, 14, 16), (6,  8, 14),
+                           (14, 16, 17), (9,  8, 14))
+    atom_zero_dihedral_values = ((0, 1, 4, 5), (0, 1, 4, 6))
+    atom_i_dihedral_values = ((4, 6, 8, 14), (6, 8, 14, 15),
+                              (6, 8, 14, 16), (7, 6, 8, 14),
+                              (8, 14, 16, 17), (8, 14, 16, 18),
+                              (9, 8, 14, 15), (9, 8, 14, 16),
+                              (10, 8, 14, 15), (10, 8, 14, 16),
+                              (11, 10, 8, 14), (12, 10, 8, 14),
+                              (13, 10, 8, 14), (14, 16, 18, 19),
+                              (14, 16, 18, 20), (14, 16, 18, 21),
+                              (15, 14, 16, 17), (15, 14, 16, 18))
+    atom_zero_improper_values = ()
+    atom_i_improper_values = ((8, 16, 14, 15), (14, 18, 16, 17))
+
+
+class TestPRMNCRST(TOPBase):
+    # Test case of PARM7 with no non-hydrogen including dihedrals
+    ref_filename = PRMNCRST
+    expected_n_atoms = 6
+    expected_n_residues = 1
+    ref_proteinatoms = 6
+    expected_n_bonds = 5
+    expected_n_angles = 7
+    expected_n_dihedrals = 3
+    expected_n_impropers = 0
+    atom_i = 4
+    expected_n_zero_bonds = 1
+    expected_n_i_bonds = 2
+    expected_n_zero_angles = 3
+    expected_n_i_angles = 4
+    expected_n_zero_dihedrals = 1
+    expected_n_i_dihedrals = 3
+    expected_n_zero_impropers = 0
+    expected_n_i_impropers = 0
+    atom_zero_bond_values = ((0, 1),)
+    atom_i_bond_values = ((1, 4), (4, 5))
+    atom_zero_angle_values = ((0, 1, 2), (0, 1, 3), (0, 1, 4))
+    atom_i_angle_values = ((0, 1, 4), (1, 4, 5), (2, 1, 4), (3, 1, 4))
+    atom_zero_dihedral_values = ((0, 1, 4, 5),)
+    atom_i_dihedral_values = ((0, 1, 4, 5), (2, 1, 4, 5), (3, 1, 4, 5))
+    atom_zero_improper_values = ()
+    atom_i_improper_values = ()
+
+
+class CheckErrors(object):
```

Comment:
@IAlibay I don't think these tests will be run.  pytest looks for classes which are named `class TestSomething` or functions/methods called `test_thing`

Each of these three files should be an individual test, so the class should look like:

```python
class TestErrors(object):
    def test_missing_version(self):
        with pytest.raises etc

    def test_missing_title(self):
        etc

Reply:
Sorry about that, looks like I need to have a more careful read of the pytest docs. It should be fixed in 07adc38